js/interface/add-markers.js

```diff
@@ -0,0 +1,500 @@
+        return {
+          id: marker.id,
+          data: {
+            name: marker.data[this.markerNameColumn],
+            floor: marker.data[this.markerFloorColumn],
+            type: marker.data[this.markerTypeColumn],
+            icon: markerData ? markerData.icon : '',
+            color: markerData ? markerData.color : '#333333',
+            size: markerData ? markerData.size : '24px',
+            positionx: marker.data[this.markerXPositionColumn],
+            positiony: marker.data[this.markerYPositionColumn],
+            updateName: false,
+            copyOfName: ''
+          }
+        }
+      })
+
+      return newMarkerData
+    },
+    setActiveMarker(index, forced) {
+      if (this.activeMarker !== index || forced) {
+        this.activeMarker = index
+        this.setupFlPanZoom()
+      }
+    },
+    updateFloor(floorName, index) {
+      this.mappedMarkerData[index].data.floor = floorName
+      this.saveDebounced()
+      this.setupFlPanZoom()
+    },
+    updateMarker(marker, index) {
+      this.mappedMarkerData[index].data.type = marker.name
+      this.mappedMarkerData[index].data.icon = marker.icon
+      this.mappedMarkerData[index].data.color = marker.color
+      this.mappedMarkerData[index].data.size = marker.size
+      this.saveDebounced()
+      this.setupFlPanZoom()
+    },
+    toUpdateName(index, currentName) {
+      this.mappedMarkerData[index].data.updateName = !this.mappedMarkerData[index].data.updateName
+      this.mappedMarkerData[index].data.copyOfName = currentName
+      this.$nextTick(() => this.$refs['changename-' + index][0].focus())
+    },
+    confirmName(index, fromCancel) {
+      this.mappedMarkerData[index].data.updateName = !this.mappedMarkerData[index].data.updateName
+
+      if (!fromCancel) {
+        this.saveDebounced()
+        this.setupFlPanZoom()
+      }
+    },
+    cancelNameUpdate(index) {
+      this.mappedMarkerData[index].data.name = this.mappedMarkerData[index].data.copyOfName
+      this.mappedMarkerData[index].data.copyOfName = ''
+      this.confirmName(index, true)
+    },
+    deleteMarker(index) {
+      const markers = this.flPanZoomInstance.markers.getAll()
+      const markerId = $('.floor-wrapper-holder')
+        .find('.marker[data-name="' + this.mappedMarkerData[index].data.name + '"]')
+        .attr('id')
+        
+      if (markerId) {
+        this.flPanZoomInstance.markers.remove(markerId, {keepInDom: false})
+      }
+
+      this.mappedMarkerData.splice(index, 1)
+      this.setActiveMarker(0, true)
+      this.saveDebounced()
+    },
+    nameWithId({ name, id }) {
+      return `${name} — [${id}]`
+    },
+    createNewData() {
+      const name = `${this.appName} - Markers`
+
+      Fliplet.Modal.prompt({
+        title: 'Please type a name for your data source:',
+        value: name
+      }).then((name) => {
+        if (name === null || name === '') {
```

Comment:
`null` and `''` should be dealt with differently.

- `null` - user cancelled, nothing happens
- `''` user entered no name, either provide a default and continue, or ask the user enter again